### DISCUSSION TITLE
Subclipse 4.3.2 released

### DISCUSSION THREAD

**Discussion by markphip** [subclipse/subclipse#150] [General]
This was not a big release, it was mainly just triggered by moving the build process over to Github Actions and needing to make sure the entire process works. That said, the development snapshot version has supported Subversion 1.14 for quite a while now so this has been pushed to the official release and was long overdue.  I also was able to get SVNKit updated to the latest version. There are some fixes that come in as well.

Details are here: https://github.com/subclipse/subclipse/releases/tag/4.3.2

Mark

**Comment by dbarrieu** [comment]
Hi,
Today our target plateform is dowm. https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/ is not accessible.
Last buil OK was the 08/11. 

The fellowing message is displayed by Maven :: 

[INFO] Adding repository https://dl.bintray.com/subclipse/releases/subclipse/4.3.x
[INFO] Fetching p2.index from https://dl.bintray.com/subclipse/releases/subclipse/4.3.2/ (172B)
[ERROR] Failed to resolve target definition D:\jenkins\workspace\AA_Caddie_Integration_Builds_win64_x86\bin\fractiaactidiagrcp\fr.actia.actidiag.targetdefinition\fr.actia.actidiag.targetdefinition.target: Could not find "org.tigris.subversion.subclipse.feature.group/4.3.0.201901172050" in the repositories of the current location -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException

Is this update can be the cause if this issue ?

**Comment by markphip** [comment]
It is true that the 4.3.0 version of the feature would not be available at the 4.3.x update site URL since the current version is now 4.3.2.  You could get it by using the 4.3.0 update site .. or you could move to the 4.3.2 version of the feature.

So you will need to adjust your build. Either pull in the new version of the feature or adjust the URL you are using for the update site to be the specific 4.3.0 URL.  https://dl.bintray.com/subclipse/releases/subclipse/4.3.0/  You should note though that the more specific URL only includes Subclipse and not its dependencies.  The 4.3.x version is a "meta" update site that pulls in Subclipse and the dependencies. So the best thing to do would be to move to the 4.3.2 version in your build process.

**Comment by dbarrieu** [comment]
Hi,

Same problem with 4.3.2 : 

[INFO] Fetching p2.index from https://dl.bintray.com/subclipse/releases/subclipse/4.3.2/ (172B)
[INFO] Adding repository https://dl.bintray.com/subclipse/releases/subclipse/4.3.2
[ERROR] Failed to resolve target definition D:\jenkins\workspace\AA_Caddie_Integration_Builds_win64_x86\bin\fractiaactidiagrcp\fr.actia.actidiag.targetdefinition\fr.actia.actidiag.targetdefinition.target: Could not find "org.tigris.subversion.clientadapter.javahl.feature.feature.group/1.11.1.201901162041" in the repositories of the current location -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException

**Comment by markphip** [comment]
Well it is not the "same problem" now it is complaining about JavaHL.

The 1.11 version is there, but there is also a 1.14 version.  In the Eclipse UI you can ask it to show all versions and choose an older version but it shows the 1.14 (newest) version by default.  I do not know what Maven offers here so I cannot help you with that.  You definitely should not be using 1.11 though as it is not a supported SVN version.  1.10 and 1.14 are the current LTS versions and the only ones currently supported.  All of them are available.

**Comment by dbarrieu** [comment]
So in my target plateforme i must add the URL https://dl.bintray.com/subclipse/releases/javahl/1.14.0/ ? Only with the URL https://dl.bintray.com/subclipse/releases/subclipse/4.3.2/ it's not working.

[INFO] Fetching p2.index from https://dl.bintray.com/subclipse/releases/subclipse/4.3.2/ (172B)
[INFO] Adding repository https://dl.bintray.com/subclipse/releases/subclipse/4.3.2
[ERROR] Failed to resolve target definition D:\jenkins\workspace\AA_Caddie_Integration_Builds_win64_x86\bin\fractiaactidiagrcp\fr.actia.actidiag.targetdefinition\fr.actia.actidiag.targetdefinition.target: Could not find "org.tigris.subversion.clientadapter.javahl.feature.feature.group/1.14.0.202000511413" in the repositories of the current location -> [Help 1]

**Reply by markphip** [reply]
So is your an app an Eclipse RCP app that you are building via Maven and Tycho?  I have one of these (it does not pull in Subclipse though) but looking at other features it pulls in, they seem to be specified in a "targetplatform" file and have XML that looks like this?

```
<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="false" type="InstallableUnit">
<unit id="com.collabnet.desktop.feature.feature.group"/>
<unit id="com.collabnet.ee.reports.feature.feature.group"/>
<unit id="com.collabnet.gerrit.feature.feature.group"/>
<unit id="com.collabnet.gerrit.workflow.feature.feature.group"/>
<unit id="com.collabnet.git.feature.feature.group"/>
<repository location="https://dl.bintray.com/collabnet/snapshots/desktop/master/"/>
</location>
```

We never have to make updates when we post new versions.  I wonder if you have something equivalent to this where you added version specifiers so your build will only work if the exact version is available?  Without the specifier it would grab the latest version. If that is the case, I would suggest dropping the version specifiers?

I am just speculating here though.

**Comment by markphip** [comment]
You are only giving me the errors so it just leave me to guess what you are doing in your code.  But somewhere you had a reference to the 4.3.x update site URL. I think you should leave that as is.  Somewhere else you specify to install version 4.3.0 of the subclipse feature. I think you should update that to 4.3.2.  Or maybe your code allows something like a greaterThanOrEqual specifier and you could use that instead of an exact version?

Likewise, you are specifying 1.11 of JavaHL and you should update that to 1.14.  But all of these versions are available on the 4.3.x update site and ought to work if specified correctly.

**Comment by dbarrieu** [comment]
Hi,
We are using Maven and Tycho to build an Eclipse RCP app. Our target is define with the following location

<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
		  <repository location="https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/"/>
		  <unit id="org.tigris.subversion.clientadapter.javahl.feature.feature.group" version="1.11.1.201901162041"/>
		  <unit id="org.tigris.subversion.clientadapter.svnkit.feature.feature.group" version="1.8.12.3"/>
		  <unit id="org.tigris.subversion.subclipse.feature.group" version="4.3.0.201901172050"/>
		  <unit id="org.tigris.subversion.subclipse.graph.feature.feature.group" version="4.2.0.1"/>
		  <unit id="org.tigris.subversion.subclipse.mylyn.feature.feature.group" version="4.2.0.1"/>
		  <unit id="org.tmatesoft.svnkit.feature.group" version="1.8.12.r10533_v20160129_0158"/>
</location>

We are using Java 11 and Eclipse 06-2020 and during the Jenkins build this new issue suddenly appeared the 11/12 (working well the 08/12) :

[INFO] Adding repository https://dl.bintray.com/subclipse/releases/subclipse/4.3.x
[INFO] Fetching p2.index from https://dl.bintray.com/subclipse/releases/subclipse/4.3.2/ (172B)
[ERROR] Failed to resolve target definition D:\jenkins\workspace\AA_Caddie_Integration_Builds_win64_x86\bin\fractiaactidiagrcp\fr.actia.actidiag.targetdefinition\fr.actia.actidiag.targetdefinition.target: Could not find "org.tigris.subversion.subclipse.feature.group/4.3.0.201901172050" in the repositories of the current location -> [Help 1]

Regards.

**Comment by dbarrieu** [comment]
following location sorry 

<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
		  <repository location="https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/"/>
		  <unit id="org.tigris.subversion.clientadapter.javahl.feature.feature.group" version="1.11.1.201901162041"/>
		  <unit id="org.tigris.subversion.clientadapter.svnkit.feature.feature.group" version="1.8.12.3"/>
		  <unit id="org.tigris.subversion.subclipse.feature.group" version="4.3.0.201901172050"/>
		  <unit id="org.tigris.subversion.subclipse.graph.feature.feature.group" version="4.2.0.1"/>
		  <unit id="org.tigris.subversion.subclipse.mylyn.feature.feature.group" version="4.2.0.1"/>
		  <unit id="org.tmatesoft.svnkit.feature.group" version="1.8.12.r10533_v20160129_0158"/>
</location>

**Comment by dbarrieu** [comment]
Ho well there is a problem on my copy / paste location !!! I try again :
<repository location="https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/"/>
<unit id="org.tigris.subversion.clientadapter.javahl.feature.feature.group" version="1.11.1.201901162041"/>
<unit id="org.tigris.subversion.clientadapter.svnkit.feature.feature.group" version="1.8.12.3"/>
<unit id="org.tigris.subversion.subclipse.feature.group" version="4.3.0.201901172050"/>
<unit id="org.tigris.subversion.subclipse.graph.feature.feature.group" version="4.2.0.1"/>
<unit id="org.tigris.subversion.subclipse.mylyn.feature.feature.group" version="4.2.0.1"/>
<unit id="org.tmatesoft.svnkit.feature.group" version="1.8.12.r10533_v20160129_0158"/>

**Comment by dbarrieu** [comment]
XML syntax seems to be not supported  :

repository location="https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/"
unit id="org.tigris.subversion.clientadapter.javahl.feature.feature.group" version="1.11.1.201901162041"
unit id="org.tigris.subversion.clientadapter.svnkit.feature.feature.group" version="1.8.12.3"
unit id="org.tigris.subversion.subclipse.feature.group" version="4.3.0.201901172050"
unit id="org.tigris.subversion.subclipse.graph.feature.feature.group" version="4.2.0.1"
unit id="org.tigris.subversion.subclipse.mylyn.feature.feature.group" version="4.2.0.1"
unit id="org.tmatesoft.svnkit.feature.group" version="1.8.12.r10533_v20160129_0158"

**Reply by markphip** [reply]
It is Markdown so just surround it with a code fence.  Your problem is exactly what I described in an earlier message ... that you need to drop the `version="x.y.z"` from your file. This is what I think should be in the location block within your file:

```
<repository location="https://dl.bintray.com/subclipse/releases/subclipse/4.3.x/"/>
<unit id="org.tigris.subversion.clientadapter.javahl.feature.feature.group"/>
<unit id="org.tigris.subversion.clientadapter.svnkit.feature.feature.group"/>
<unit id="org.tigris.subversion.subclipse.feature.group"/>
<unit id="org.tigris.subversion.subclipse.graph.feature.feature.group"/>
<unit id="org.tigris.subversion.subclipse.mylyn.feature.feature.group"/>
<unit id="org.tmatesoft.svnkit.feature.group"/>
```

**Reply by dbarrieu** [reply]
Hi mark,

Well if i done this, i have now a problem with the Java environment for svnkit feature.

[ERROR] Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST D:\aa_workarea\.m2\repository\p2\osgi\bundle\org.tmatesoft.sqljet\1.1.13.r1341\org.tmatesoft.sqljet-1.1.13.r1341.jar!/META-INF/MANIFEST.MF: Unknown OSGi execution environment: 'JavaSE-1.5' -> [Help 1]

Regars.

**Comment by dbarrieu** [comment]
Problem is fixed. Thanks for your help.
